src/adaptations/device-layer/NetworkTelemetryManager.cpp

```diff
@@ -109,14 +110,18 @@ WEAVE_ERROR NetworkTelemetryManager::Init(void)
     return err;
 }
 
-
 #if WEAVE_DEVICE_CONFIG_ENABLE_WIFI_TELEMETRY
 void WiFiTelemetry::GetTelemetryStatsAndLogEvent(void)
 {
     WEAVE_ERROR err;
 
-    err = ConnectivityMgr().GetAndLogWifiStatsCounters();
-    SuccessOrExit(err);
+    if (ConnectivityMgr().IsWiFiStationProvisioned() &&
+        ConnectivityMgr().IsWiFiStationEnabled() &&
+        ConfigurationMgr().IsPairedToAccount())
```

Comment:
I'm going to accept this, but I'm still really uncertain about the IsPairedToAccount() logic.  Remind me the rationale for not logging stats when the device isn't paired to an account; in particular, what goes wrong if we do?

Reply:
@suryanshup I agree that capturing the answer to Jay's question this is important.

If the proper fix is being deferred, would it make sense to create a github issue with the explanation and reference it with a `//TODO(#issue)` comment?

Reply:
Based on an offline conversation I had with Jay, we agreed that we should remove the service provisioning check. I will fix that in a separate PR